addons/account/models/account_move.py

```diff
@@ -5299,11 +5467,14 @@ def reconcile(self):
 
         # ==== Create partials ====
 
-        partials = self.env['account.partial.reconcile'].create(sorted_lines._prepare_reconciliation_partials())
-
-        # Track newly created partials.
+        partials = sorted_lines._create_reconciliation_partials()
```

Comment:
is there a real reason for that change? the diff is messed up and it's hard to see the difference between the old implementation and yours

Reply:
I need to create all partials at the same time to avoid performances issues (orm invalidation + recomputation of residual amounts). But I also need to create the exchange difference and link them to the corresponding partials. That's the reason why I made change `_prepare_reconciliation_partials` to `_create_reconciliation_partials`: to create the partials and then, link the exchange difference journal entries